satpy/enhancements/__init__.py

```diff
@@ -134,23 +92,46 @@ def _enhance_whole_array(func, data, bands, exclude, pass_dask, use_map_blocks):
     return data
 
 
-def _call_enh_func(func, band_data_arr, pass_dask, use_map_blocks, extra_kwargs):
-    if pass_dask:
-        dims = band_data_arr.dims
-        coords = band_data_arr.coords
-        if use_map_blocks:
-            d_arr = da.map_blocks(func,
-                                  band_data_arr.data,
-                                  meta=np.array((), dtype=band_data_arr.dtype),
-                                  dtype=band_data_arr.dtype,
-                                  chunks=band_data_arr.chunks,
-                                  **extra_kwargs)
-        else:
-            d_arr = func(band_data_arr.data, **extra_kwargs)
-        band_data_arr = xr.DataArray(d_arr, dims=dims, coords=coords)
-    else:
-        band_data_arr = func(band_data_arr, **extra_kwargs)
-    return band_data_arr
+def on_separate_bands(func):
```

Comment:
Does the order of this decorator with the other dask decorators matter? If so, can it be documented here.

Reply:
Yes it does, since `on_dask_array` for example make the decorated function act on a dask array as the docstring says.
I could write in the `on_separate_bands` docstring that it acts on DataArrays, ok?

Reply:
That would help, but you could even go as far as saying "If this is used in combination with on_dask_array...., then it should be used *after* those decorators. For example:" and then show the combination of the two decorators. Just thinking about the emails I'm going to have to answer when someone gets it wrong.

Reply:
done